src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -710,6 +710,15 @@ std::string network::get_primitive_info(const primitive_id& id) const {
     return node.type()->to_string(node);
 }
 
+cldnn::impl_types network::get_impl_type(const primitive_id& id) const {
+    auto prim_inst = find_primitive(id);
+
+    OPENVINO_ASSERT(prim_inst, "[GPU] Can't get implementation type, since topology",
+                               "doesn't contain primitive with requested id: ", id);
+
+    return prim_inst->get_node().get_preferred_impl_type();
```

Comment:
I think it's better to use `impl->is_cpu()`, as program_node may be unavailable in deserialized network.